staging/src/k8s.io/legacy-cloud-providers/aws/aws.go

```diff
@@ -1521,6 +1533,27 @@ func (c *Cloud) NodeAddresses(ctx context.Context, name types.NodeName) ([]v1.No
 			}
 		}
 
+		// IPv6. Ordered after IPv4 addresses, so legacy code can continue to just use the "first" address in a dual-stack cluster.
+		for _, macID := range macIDs {
+			ipPath := path.Join("network/interfaces/macs/", macID, "ipv6s")
+			ips, err := c.metadata.GetMetadata(ipPath)
+			if err != nil {
+				if isAWSNotFound(err) {
+					// No IPv6 configured. Not an error, just a disappointment.
+					continue
+				}
+				return nil, fmt.Errorf("error querying AWS metadata for %q: %q", ipPath, err)
+			}
+
+			for _, ip := range strings.Split(ips, "\n") {
+				if ip == "" {
```

Comment:
I remember to discuss with one person from AWS before about IPs in non canonical forms, should we canonicalize the IPv6 address?

I can try to find the discussion later if that can help

Reply:
canonicalize means parse the ip and output the string format from the golang stdlib `netutils.ParseIPSloppy(ip).String()`

Reply:
oh, wait, forget it, I think that the problem is if you canonicalize this is going to hotloop, scratch this comment